modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/FieldBase/ReactFieldBase.es.js

```diff
@@ -236,6 +227,29 @@ export function FieldBase({
 		type === 'paragraph' ||
 		type === 'radio';
 	const showPopover = fieldName === 'inputMaskFormat';
+	const showFor =
+		type === 'text' ||
+		type === 'numeric' ||
+		type === 'image' ||
+		type === 'search_location';
+
+	const fieldDetails = getFieldDetails({
+		errorMessage,
+		hasError,
+		label,
+		required,
+		text,
+		tip,
+		warningMessage,
+	});
+
+	const fieldDetailsId = `${id ?? name}_fieldDetails`;
+
+	const accessibleProps = {
+		...(accessible && fieldDetails && {'aria-labelledby': fieldDetailsId}),
+		...(showFor && {htmlFor: id ?? name}),
+		...(!showFor && {tabIndex: 0}),
```

Comment:
Could you please use ternary conditional to avoid code repetition?

```suggestion
		...(showFor ? {htmlFor: id} : {tabIndex: 0}),
```

Reply:
Hi @alinedoleron,

The problem is that ID is not defined for text fields, so if we don't use name as well, then we have no ID.

How can we fix that?

Regards.